psutil/_psutil_common.c

```diff
@@ -40,7 +40,7 @@ PyUnicode_DecodeFSDefaultAndSize(char *s, Py_ssize_t size) {
  * If msg != "" the exception message will change in accordance.
  */
 PyObject *
-NoSuchProcess(char *msg) {
+NoSuchProcess(const char *msg) {
```

Comment:
why is this necessary?

Reply:
`const char *` means pointer to const characters. The const modifier signals the caller that NoSuchProcess will not change the passed message. Otherwise, it's not clear and one has to inspect the function body. In case someone accidentally writes to msg using that const pointer it's a compile error. This change also remove compiler warnings since NoSuchProcess is called with string literals that effectively have the type `const char *`. Thus, this additional type safety protects against accidental writing into a const character array through a non-const pointer which would invoke undefined behavior (e.g. crash if the string literals are placed in a read-only data section).